core/clustersmngr/factory.go

```diff
@@ -542,6 +584,13 @@ func (cf *clustersManager) userNsList(ctx context.Context, user *auth.UserPrinci
 		return userNamespaces
 	}
 
+	if cf.useUserClientForNamespaces {
+		err := cf.updateNamespacesFromUserClient(ctx, user)
+		if err != nil {
+			cf.log.Error(err, "failed updating namespaces from user client")
+		}
```

Comment:
We could call out why we don't return an error here?

Reply:
:thumbsup: